include/boost/geometry/algorithms/detail/disjoint/multirange_geometry.hpp

```diff
@@ -68,13 +68,11 @@ struct multirange_constant_size_geometry
                         typename boost::range_value<MultiRange>::type,
                         ConstantSizeGeometry
                     >
-            > unary_predicate_type;
+            >;
 
-        return detail::check_iterator_range
-            <
-                unary_predicate_type
-            >::apply(boost::begin(multirange), boost::end(multirange),
-                     unary_predicate_type(constant_size_geometry, strategy));
+        auto const end = boost::end(multirange);
+        return std::find_if(boost::begin(multirange), end, 
+            unary_predicate_type(constant_size_geometry, strategy)) == end;
```

Comment:
I know that this is how this type was called before but don't you think that it'd be more readable if it instead of generic `predicate` we'd call it `not_disjoint`? Then this expression/sentence would be: find if not disjoint and the reader wouldn't have to check what the "predicate" is.